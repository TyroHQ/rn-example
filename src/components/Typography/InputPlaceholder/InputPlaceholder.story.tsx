import { storiesOf } from "@storybook/react-native";
import figmaDecorator from "storybook-addon-figma";

import * as usage from "./InputPlaceholder.usage";

let figma = undefined;

let story = storiesOf("InputPlaceholder", module);

if (figma) {
  story = story.addDecorator(
    figmaDecorator({
      url: figma
    })
  );
}

for (const example in usage) {
  story = story.add(example, usage[example]);
}
