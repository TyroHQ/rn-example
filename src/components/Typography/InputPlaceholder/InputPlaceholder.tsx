import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";

const InputPlaceholderContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class InputPlaceholder extends React.Component<{}> {
  static defaultProps: InputPlaceholder["props"];
  render() {
    return <Text>InputPlaceholder</Text>;
  }
}
