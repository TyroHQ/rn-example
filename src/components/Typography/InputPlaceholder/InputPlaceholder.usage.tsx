import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { InputPlaceholder } from "./";

export const basic = () => <InputPlaceholder />;
