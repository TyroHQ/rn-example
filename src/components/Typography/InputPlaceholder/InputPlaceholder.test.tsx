import { shallow } from "enzyme";
import * as usage from "./InputPlaceholder.usage";

describe("<InputPlaceholder />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
