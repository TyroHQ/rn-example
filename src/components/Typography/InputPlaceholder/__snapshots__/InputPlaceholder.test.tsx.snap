// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPlaceholder /> renders well in basic mode 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  InputPlaceholder
</Text>
`;
